### ISSUE TITLE
setting defaults for tokens

### ISSUE THREAD

**Issue by spoore1** [SatelliteQE/testimony#116]
It would be useful to be able to set default vaults for tokens that would apply to all test cases when testimony runs.   If I have a test suite with a lot of test cases where several tokens are set identically, it would be helpful to set that outside the tests.  

A config file alongside the tests maybe?  Or command line options?  Or maybe a new token to pull that info from a config file?

Keeping things flexible, if a token is defined in a config, it would be overridden if defined in a test case.

Does that make sense?

**Comment by elyezer** [comment]
@spoore1 defining the tokens on the module level does not help? Or you are looking for a solution that will be global so every module would inherit it?

**Comment by spoore1** [comment]
Can I do that?  Add a docstring for the module with tokens?

Yeah, actually, that's better.   Tokens at the module and class level would both be very useful I think.  At the moment, I see them listed as Unexpected tokens:

```
# cat test_2.py 
"""
@casecomponent: ipa
@requirement: My test Req
"""

class TestExample(object):
    def test_case_0001(self):
        """ Test Case 0001 """
        pass

    def test_case_0002(self):
        """ Test Case 0002 """
        pass

    def test_case_0003(self):
        """ Test Case 0003 """
        pass
# testimony print test_2.py 
test_2.py
=========

test_case_0001
--------------

Test:
 Test Case 0001

Unexpected tokens:
  Casecomponent: ipa
  Requirement: My test Req

test_case_0002
--------------

Test:
 Test Case 0002

Unexpected tokens:
  Casecomponent: ipa
  Requirement: My test Req

test_case_0003
--------------

Test:
 Test Case 0003

Unexpected tokens:
  Casecomponent: ipa
  Requirement: My test Req
```

**Comment by omaciel** [comment]
@spoore1 I believe what you want is pass `--tokens` and/or `--minimum-tokens` to `testimony` like we're doing in [Robottelo](https://github.com/SatelliteQE/robottelo/blob/a498e06e3380a2b0fc9eec2b8c98d5bc8f14ce76/Makefile#L57)

**Comment by spoore1** [comment]
@omaciel you are correct.  I had misunderstood the help listing for that.  Looks like that works now:
```
# testimony print test_2.py --tokens="id, description, test, casecomponent, requirement" --minimum-tokens="test"
test_2.py
=========

test_case_0001
--------------

Casecomponent:
 ipa

Requirement:
 My test Req

Test:
 Test Case 0001


test_case_0002
--------------

Casecomponent:
 ipa

Requirement:
 My test Req

Test:
 Test Case 0002


test_case_0003
--------------

Casecomponent:
 ipa

Requirement:
 My test Req

Test:
 Test Case 0003
```

That's awesome.  I think this can be closed.    Is this in the doc?  If not, might be a good example to add.

Thanks,
Scott

**Comment by omaciel** [comment]
@spoore1 not sure but if not we should definitely document it. @sthirugn ?

**Comment by elyezer** [comment]
We need to clarify the documentation

**Comment by sthirugn** [comment]
This is briefly explained in http://testimony-qe.readthedocs.io/en/latest/#test-case-docstring-format.  But examples are not shown on how to override the defaults.  +1 to adding examples.

This can be addressed in https://github.com/SatelliteQE/testimony/issues/119.  

@spoore1: If you want to contribute to this doc, please feel free to.  Otherwise it will be worked eventually.
